enhancements/workload-partitioning/cpu-limits-workload-partitioning.md

```diff
@@ -0,0 +1,417 @@
+---
+title: cpu-limits-workload-partitioning
+authors:
+  - "@eggfoobar"
+reviewers:
+  - "@jerpeter1"
+  - "@mrunalp"
+  - "@rphillips"
+  - "@browsell"
+  - "@haircommander"
+  - "@MarSik"
+  - "@Tal-or"
+approvers:
+  - "@jerpeter1"
+  - "@mrunalp"
+api-approvers:
+  - "None"
+creation-date: 2024-01-24
+last-updated: 2024-01-24
+tracking-link:
+  - https://issues.redhat.com/browse/OCPEDGE-57
+see-also:
+  - "/enhancements/workload-partitioning"
+---
+
+# CPU Limits for Workload Partitioning
+
+## Summary
+
+This enhancements builds on top of the [Management Workload
+Partitioning](management-workload-partitioning.md) enhancement to provide the
+ability for workload partitioning to take into account CPU limits during Pod
+admission. Currently only CPU requests are used during Pod admission and any Pod
+that uses CPU limits is ignored. With this change the Pod admission webhook will
+take into account CPU limits and use the existing mechanism to pass the CPU
+limits information to the underlying container runtime.
+
+## Motivation
+
+Workload partitioning currently does not support mutating containers that have
+CPU limits. The original premise was that all OCP Pods did not set limits, if
+limits were present then the default behavior would set the CPU `requests.cpu`
+to the limit value and thus getting in the way of the scheduler to use the
+`cpushares` so it was decided to avoid modifying those pods.
+([commit](https://github.com/openshift/kubernetes/commit/c6395e702e5f02c11ebb7659a18cef0b24609bfb)).
+However it has been found that at least one cluster container does set limits. A
```

Comment:
out of curiosity: which one?

Reply:
This feature was written up a while ago, I don't recall exactly, @browsell do you recall which one of the platform pods was found to have this?

Reply:
Yes, I think I am curious about the workload, too. In general we try to avoid using cpu limits on OpenShift/k8s so from that perspective we would want to know whether there is a real need for this feature.

Reply:
It was the klusterlet pod which has since been mitigated. The main driver for this is to allow limits on some day2 operators which do support limits (ex. logging)

Reply:
Thanks Brent, resolving this comment for now